Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/DriverInformation.hs

```diff
@@ -101,6 +102,34 @@ updateEnabledVerifiedState driverId isEnabled isVerified = do
         <> [DriverInformationLastEnabledOn =. val (Just now) | isEnabled]
     where_ $ tbl ^. DriverInformationDriverId ==. val (toKey $ cast driverId)
 
+updateDynamicBlockedState :: Id Person.Driver -> Maybe Text -> Maybe Int -> Bool -> SqlDB ()
+updateDynamicBlockedState driverId blockedReason Nothing isBlocked = do
```

Comment:
we don't need to have this case match thing here ? this way once we have blocked someone for some time we would not be able to block them indefinitely (i.e. set `DriverInformationBlockExpiryTime` back to Nothing), even though we can give a longer time but still once written we cannot make it NOTHING from dashboard.

Reply:
instead we can just have the below definition of the function and do:

`(flip addUTCTime now . (* 3600)) <$> blockedExpiryTime`
instead of 
`addUTCTime (maybe 0 (\x -> fromIntegral x * 60 * 60) blockedExpiryTime) now` ?

Reply:
no we can do that like block driver and  keep the block expiry time as null

Reply:
so it can only be unblocked by the dashboard only

Reply:
let's say a driver is blocked for 3 days, and within those 3 days we want to change it to make him block forever (for some reason), can we do that through your current implementation ?

Reply:
okk understood